src/components/Imports/Providers/nextcloud/provider.js

```diff
@@ -0,0 +1,449 @@
+      const txt = await res.text().catch(() => '')
+      const err = new Error(txt || `Remote probe failed: HTTP ${res.status}`)
+      err.status = res.status
+      err.path = clean
+      throw err
+    }
+
+    const json = await res.json()
+    const data = json?.data
+
+    if (Array.isArray(data)) {
+      return {
+        kind: 'directory',
+        items: data,
+        name: clean.split('/').filter(Boolean).pop() || '/'
+      }
+    }
+
+    const name = data?.attributes?.name || data?.name || clean.split('/').pop()
+    const type = data?.attributes?.type || data?.type || 'file'
+
+    if (type === 'directory') {
+      const listing = await this.listRemote(client, accountId, clean)
+      return { kind: 'directory', items: listing, name }
+    }
+
+    return { kind: 'file', name }
+  },
+
+  async listRemote(client, accountId, path = '/') {
+    const clean = this._joinRemotePath(path)
+    const encodedPath = this._encodeRemotePath(clean)
+    const url = `/remote/nextcloud/${encodeURIComponent(
+      accountId
+    )}${encodedPath}`
+
+    const res = await client.stackClient.fetch('GET', url)
+
+    if (res.status === 401) {
+      const err = new Error('Unauthorized to Nextcloud')
+      err.status = 401
+      err.path = clean
+      throw err
+    }
+
+    if (res.status === 404) {
+      const txt = await res.text().catch(() => '')
+      const err = new Error(
+        txt ||
+          `${this._extractName(clean)} - 404, ${this._build404Reason(clean)}`
+      )
+      err.status = 404
+      err.path = clean
+      throw err
+    }
+
+    if (res.status !== 200) {
+      const txt = await res.text().catch(() => '')
+      const err = new Error(txt || `Remote list failed: HTTP ${res.status}`)
+      err.status = res.status
+      err.path = clean
+      throw err
+    }
+
+    const json = await res.json()
+    return json?.data || []
+  },
+
+  async findChildDirByName(client, parentId, name) {
+    const res = await client.stackClient.fetchJSON(
+      'POST',
+      '/data/io.cozy.files/_find',
+      { selector: { dir_id: parentId, name, type: 'directory' }, limit: 1 }
+    )
+    return res?.docs?.[0] || null
+  },
+
+  async createChildDir(client, parentId, name) {
+    const qs = new URLSearchParams({ Type: 'directory', Name: name })
+    const url = `/files/${encodeURIComponent(parentId)}?${qs.toString()}`
+    const res = await client.stackClient.fetch('POST', url, null, {
+      headers: { Accept: 'application/vnd.api+json' }
+    })
+
+    if (res.status >= 200 && res.status < 300) {
+      const json = await res.json().catch(() => ({}))
+      return json?.data || json
+    }
+
+    if (res.status === 409) {
+      const existing = await this.findChildDirByName(client, parentId, name)
+      if (existing) return existing
+    }
+
+    const txt = await res.text().catch(() => '')
+    throw new Error(txt || `Create directory failed: HTTP ${res.status}`)
+  },
+
+  async ensureChildDir(client, parentId, name) {
```

Comment:
don't we already have `ensureDirectoryExists` or something like that in CozyClient?